exercises/practice/rational-numbers/.meta/proof.ci.ts

```diff
@@ -49,10 +49,17 @@ export class Rational {
   }
 
   public exprational(n: number): Rational {
-    return new Rational(
-      Math.pow(this.numerator, n),
-      Math.pow(this.denominator, n)
-    )
+    if (n > 0) {
+      return new Rational(
+        Math.pow(this.numerator, n),
+        Math.pow(this.denominator, n)
+      )
+    } else {
+      return new Rational(
+        Math.pow(this.denominator, Math.abs(n)),
+        Math.pow(this.numerator, Math.abs(n))
+      )
+    }
```

Comment:
It's fine because it's a proof and not exemplar, but this would have worked too:

```typescript
return new Rational(
  Math.pow(this.denominator, Math.abs(n)),
  Math.pow(this.numerator, Math.abs(n))
)
```

The cost of the branch may, if this code was in the hot path, be as expensive as always removing the sign bit. 😁